### ISSUE TITLE
Please add support or fix the problem with TRV from Tuya TS0601 _TZE200_hue3yfsn

### ISSUE THREAD

**Issue by tyros5a** [Koenkk/zigbee-herdsman-converters#3142]
A new model from Tuya Smart Thermostat Radiator Valva does not work in iobroker. It was paired /connected with error messages:
"TS0601" not described in statesMapping.
Device xxxxx "TS0601" not described in statesMapping.

modelZigbee:TS0601
type:EndDevice
nwk:51249
manuf id:4098
manufacturer:_TZE200_hue3yfsn
power:Battery
app version:69
hard version:1
zcl version:3
stack version:0
date code:
interviewed:true
configured:false 

Example for this device:
https://www.aliexpress.com/item/1005003065409914.html?spm=a2g0s.9042311.0.0.62024c4dvojtWY 

unfortunately no one has reported here and there is no further information on how to solve the similar problem:
https://forum.iobroker.net/topic/47516/neue-tuya-thermostat-zigbee-3-0/20

Thanks in advance!!!

**Comment by PeWu** [comment]
I also have the **_TZE200_hue3yfsn** TRV. I can help with testing and debugging the device.

I also found a discussion about this TRV here: https://github.com/zigpy/zha-device-handlers/issues/1027

I can confirm the following data points:
- 2: mode (0: auto, 1: manual, 3: holiday)
- 16: heatingSetpoint (value_in_celsius * 10)

I also managed to catch this message:
```
Received Zigbee message from '[REDACTED]', type 'commandGetData', cluster 'manuSpecificTuya', data '{"data":{"data":[36,0,170,72,0,210,84,0,170,102,0,210,144,0,170,144,0,210,144,0,170,144,0,210,144,0,170,144,0,210],"type":"Buffer"},"datatype":0,"dp":109,"fn":0,"status":0,"transid":26}' from endpoint 1 with groupID 0
```
According to the other issue, 109 is the Wednesday schedule.

I tried:
- same code as `_TZE200_aoclfnxz`
- same code as `_TZE200_ckud7u2l`
- https://github.com/Lavisx/lidl-valve-z2m/blob/main/lidl_radiator_valve.js
None of them worked. The last one at least reported the set temperature (times 10).

**Comment by pirracas77** [comment]
Seems similar to this one https://www.zigbee2mqtt.io/devices/TS0601_thermostat.html

Did you try to to modify z2m files in order to make believe to z2m is the same device?

**Comment by PeWu** [comment]
Here is my initial version of a converter file for the _TZE200_hue3yfsn with only a few fields:
https://github.com/PeWu/tuya-valve-z2m/blob/main/tuya_radiator_valve.js

The setpoint, mode and child lock fields work fine. However, I'm not getting the current temperature.

**Comment by handfreezer** [comment]
My participation:

Here is the result of tuya_dump on joining step:

```
2021-10-30T00:35:26 0x2c1165fffe7c2f38 (tid:  48, s: 0) [unknown      (  2)] => 0 0
2021-10-30T00:35:26 0x2c1165fffe7c2f38 (tid:  49, s: 0) [wndow detect (  8)] => 0 False
2021-10-30T00:35:26 0x2c1165fffe7c2f38 (tid:  50, s: 0) [frost detect ( 10)] => 0 False
2021-10-30T00:35:26 0x2c1165fffe7c2f38 (tid:  51, s: 0) [unknown      ( 16)] => 0 170
2021-10-30T00:35:27 0x2c1165fffe7c2f38 (tid:  52, s: 0) [unknown      ( 24)] => 0 266
2021-10-30T00:35:27 0x2c1165fffe7c2f38 (tid:  53, s: 0) [temp calibr  ( 27)] => 0 0
2021-10-30T00:35:27 0x2c1165fffe7c2f38 (tid:  54, s: 0) [unknown      ( 31)] => 0 0
2021-10-30T00:35:27 0x2c1165fffe7c2f38 (tid:  55, s: 0) [unknown      ( 32)] => 0 170
2021-10-30T00:35:27 0x2c1165fffe7c2f38 (tid:  56, s: 0) [unknown      ( 35)] => 0 0
2021-10-30T00:35:27 0x2c1165fffe7c2f38 (tid:  57, s: 0) [child lock   ( 40)] => 0 False
2021-10-30T00:35:28 0x2c1165fffe7c2f38 (tid:  58, s: 0) [unknown      ( 45)] => 0 00000000
2021-10-30T00:35:28 0x2c1165fffe7c2f38 (tid:  59, s: 0) [unknown      ( 35)] => 0 100
2021-10-30T00:35:28 0x2c1165fffe7c2f38 (tid:  60, s: 0) [unknown      ( 46)] => 0 202101010101202101010101
2021-10-30T00:35:28 0x2c1165fffe7c2f38 (tid:  61, s: 0) [thermostat   (101)] => 0 0
2021-10-30T00:35:28 0x2c1165fffe7c2f38 (tid:  62, s: 0) [current temp (102)] => 0 21.0°C
2021-10-30T00:35:28 0x2c1165fffe7c2f38 (tid:  63, s: 0) [valve pos    (104)] => 0 210
2021-10-30T00:35:28 0x2c1165fffe7c2f38 (tid:  64, s: 0) [battery low  (105)] => 0 170
2021-10-30T00:35:28 0x2c1165fffe7c2f38 (tid:  65, s: 0) [sched mode   (107)] => 0 False
2021-10-30T00:35:29 0x2c1165fffe7c2f38 (tid:  66, s: 0) [set hst day  (110)] => 0 [  36   0 170  72   0 210  84   0 170 102   0 210 144   0 170 144   0 210 144   0 170 144   0 210 144   0 170 144   0 210 ]
2021-10-30T00:35:29 0x2c1165fffe7c2f38 (tid:  66, s: 0) [set hst day  (110)] => 0 [  36   0 170  72   0 210  84   0 170 102   0 210 144   0 170 144   0 210 144   0 170 144   0 210 144   0 170 144   0 210 ]
2021-10-30T00:35:29 0x2c1165fffe7c2f38 (tid:  67, s: 0) [set hst year (114)] => 0 [  36   0 170  72   0 210  84   0 170 102   0 210 144   0 170 144   0 210 144   0 170 144   0 210 144   0 170 144   0 210 ]
2021-10-30T00:35:30 0x2c1165fffe7c2f38 (tid:  68, s: 0) [time sync (111)] => 0 [  36   0 170  72   0 210  84   0 170 102   0 210 144   0 170 144   0 210 144   0 170 144   0 210 144   0 170 144   0 210 ]
2021-10-30T00:35:30 0x2c1165fffe7c2f38 (tid:  68, s: 0) [time sync (111)] => 0 [  36   0 170  72   0 210  84   0 170 102   0 210 144   0 170 144   0 210 144   0 170 144   0 210 144   0 170 144   0 210 ]
2021-10-30T00:35:30 0x2c1165fffe7c2f38 (tid:  69, s: 0) [sched enable (108)] => 0 [  36   0 170  72   0 210  84   0 170 102   0 210 144   0 170 144   0 210 144   0 170 144   0 210 144   0 170 144   0 210 ]
2021-10-30T00:35:31 0x2c1165fffe7c2f38 (tid:  70, s: 0) [set hst week (112)] => 0 [  36   0 170  72   0 210  84   0 170 102   0 210 144   0 170 144   0 210 144   0 170 144   0 210 144   0 170 144   0 210 ]
2021-10-30T00:35:31 0x2c1165fffe7c2f38 (tid:  70, s: 0) [set hst week (112)] => 0 [  36   0 170  72   0 210  84   0 170 102   0 210 144   0 170 144   0 210 144   0 170 144   0 210 144   0 170 144   0 210 ]
2021-10-30T00:35:31 0x2c1165fffe7c2f38 (tid:  71, s: 0) [sched set (109)] => 0 [  36   0 170  72   0 210  84   0 170 102   0 210 144   0 170 144   0 210 144   0 170 144   0 210 144   0 170 144   0 210 ]
2021-10-30T00:35:31 0x2c1165fffe7c2f38 (tid:  72, s: 0) [set hst mon  (113)] => 0 [  36   0 170  72   0 210  84   0 170 102   0 210 144   0 170 144   0 210 144   0 170 144   0 210 144   0 170 144   0 210 ]
2021-10-30T00:35:31 0x2c1165fffe7c2f38 (tid:  72, s: 0) [set hst mon  (113)] => 0 [  36   0 170  72   0 210  84   0 170 102   0 210 144   0 170 144   0 210 144   0 170 144   0 210 144   0 170 144   0 210 ]
2021-10-30T00:35:32 0x2c1165fffe7c2f38 (tid:  73, s: 0) [cur hst day  (115)] => 0 1

```

**Comment by handfreezer** [comment]
I'm working on it, having most of them fromZigbee working, but, fields are not displayed on webui. I guess it is in "exposes"  section of the converter, but I don't understand the way it is working. might you help?

**Comment by rforro** [comment]
@PeWu I was running this TRV two days straight (auto and manual mode), but it never published current temperature. Only after pairing and reseting. It's either bug in the firmware or it has to be manually requested.

**Comment by bluefish007** [comment]
@rforro 

you can try this converter based on _TZE200_e9ba97vf from moes https://github.com/bluefish007/tuya_radiator_valve/blob/main/tuya_radiator_valve.js

Following is working:
battery, comfort_temperature, current_heating_setpoint, eco_temperature, heating_stop, holiday_temperature, local_temperature, local_temperature_calibration, mode, msg_from_zigbee, preset 

But here the same bug: 
The temperature does not update itself, but if you update the local_temperature_calibration, the local_temperature updates itself too ;-)

**Comment by insipiens** [comment]
@handfreezer  I don't see those DPs match up, for instance valve position would be a percentage so 210 wouldn't be valid?

I joined mine by replacing the fingerprint for Moes TRV01 with ours in lib/moes.js and the DPs for that are:

![image](https://user-images.githubusercontent.com/68975498/140058856-5bbde018-23f3-49fa-95b3-415c77a576ba.png)

Which I'm still checking

**Comment by insipiens** [comment]
@bluefish007 The local temperature behaviour is a little odd. I note that the PR noted this issue.

I might try hooking it up to the tuya hub to see if this can be replicated on the Tuya app, unless someone else has already tried that?

**Comment by insipiens** [comment]
Been thinking about the temperature issue. I suspect its intentional to reduce battery depletion. Wireless transmission consumes a lot of power relatively.

If that is the case would you expect it to be the behaviour in z2m too?

I’ve sent a question to the vendor asking if this behaviour is intentional.

**Comment by handfreezer** [comment]
@insipiens : I suspect you're right, and that there is a get state DP maybe

**Comment by handfreezer** [comment]
@insipiens  : about valve pos : take care : in my dump , I added the DP numeric value (for what is called valve pos : it is 104). The read_tuya_dump script from koenk github is hard coded with some DP for an other device, so, we have to look only on DP numeric value, and and so 104 is matching Comfort temp on your side, so 210 is 21.0°C => sounds good

**Comment by handfreezer** [comment]
Hello,  here is where I'm for now, all data fromZigbee seems good (only week program is not implemented as I don't need...) I still have to had some toZigbee function, and expose rules, what do you think?

```
const fz = {...require('zigbee-herdsman-converters/converters/fromZigbee'), legacy: require('zigbee-herdsman-converters/lib/legacy').fromZigbee};
const tz = require('zigbee-herdsman-converters/converters/toZigbee');
const tuya = require('zigbee-herdsman-converters/lib/tuya');
const globalStore = require('zigbee-herdsman-converters/lib/store');
const exposes = require('zigbee-herdsman-converters/lib/exposes');
const reporting = require('zigbee-herdsman-converters/lib/reporting');
const e = exposes.presets;
const ea = exposes.access;

const tuyaLocal = {
    dataPoints: {
        zsMode: 2,
        zsWindowCheck: 8,
        zsFrostProtection: 10,
        zsHeatingSetpoint: 16,
        zsCurrentTemp: 24,
        zsCorrectionTemp: 27,
        zsWorkDays: 31,
        zsHolidaysSetpoint: 32,
        zsBatteryPercentage: 35,
        zsChildLock: 40,
        zsFaulty: 45,
        zsHolidaySet: 46,
        zsKSSW: 101,
        zsKCWD: 102,
        zsSSWD: 104,
        zsJLWD: 105,
        zsHeatingStopping: 107,
        zsProgramMonday: 108,
        zsProgramWednesday: 109,
        zsProgramFriday: 110,
        zsProgramSunday: 111,
        zsProgramTuesday: 112,
        zsProgramThursday: 113,
        zsProgramSaturday: 114,
        zsOnline: 115,
    },
    thermostatWeekFormat: {
        0: '5+2',
        1: '6+1',
        2: '7',
    },
    thermostatOnlineStatus: {
        'online': 'online',
        'offline': 'offline',
        0: 'offline',
        1: 'online',
    },
    thermostatOnOff: {
        0: 'OFF',
        'OFF': 0,
        1: 'ON',
        'ON': 1,
    },
};

const fzLocal = {
    zs_thermostat: {
        cluster: 'manuSpecificTuya',
        type: ['commandGetData', 'commandSetDataResponse'],
        convert: (model, msg, publish, options, meta) => {
            const dp = msg.data.dp;
            const value = tuya.getDataValue(msg.data.datatype, msg.data.data);

            switch (dp) {
            case tuyaLocal.dataPoints.zsMode:
                switch (value) {
                case 0: // auto
                    return {away_mode: 'OFF', preset: 'schedule'};
                case 1: // manual
                    return {away_mode: 'OFF', preset: 'manual'};
                case 3: // holiday
                    return {away_mode: 'ON', preset: 'holiday'};
                default:
                    meta.logger.warn('zigbee-herdsman-converters:zsThermostat: ' +
                        `preset ${value} is not recognized.`);
                    break;
                }
                break;
            case tuyaLocal.dataPoints.zsWindowCheck:
                return {window_detection: tuyaLocal.thermostatOnOff[value]};
            case tuyaLocal.dataPoints.zsFrostProtection:
                return {frost_protection: tuyaLocal.thermostatOnOff[value]};
            case tuyaLocal.dataPoints.zsHeatingSetpoint:
                return {current_heating_setpoint: (value / 10).toFixed(1)};
            case tuyaLocal.dataPoints.zsCurrentTemp:
                return {local_temperature: (value / 10).toFixed(1)};
            case tuyaLocal.dataPoints.zsCorrectionTemp:
                return {local_temperature_calibration: parseFloat((value / 10).toFixed(1))};
            case tuyaLocal.dataPoints.zsWorkDays:
                meta.logger.debug('converter TS0601 => zsWorkDays : dp=' + dp + ' data=' + JSON.stringify(msg.data));
                meta.logger.debug('converter TS0601 => zsWorkDays : result value=' + JSON.stringify(msg.data));
                return {week: tuyaLocal.thermostatWeekFormat[value]};
            case tuyaLocal.dataPoints.zsHolidaysSetpoint:
                meta.logger.debug('converter TS0601 => zsHolidaysSetpoint : dp=' + dp + ' data=' + JSON.stringify(msg.data));
                meta.logger.debug('converter TS0601 => zsHolidaysSetpoint : result value=' + JSON.stringify(msg.data));
                return {holiday_set_point: tuyaLocal.thermostatOnOff[value]};
            case tuyaLocal.dataPoints.zsBatteryPercentage:
                meta.logger.debug('converter TS0601 => zsBatteryPercentage : dp=' + dp + ' data=' + JSON.stringify(msg.data));
                meta.logger.debug('converter TS0601 => zsBatteryPercentage : result value=' + value);
                return {battery: value};
            case tuyaLocal.dataPoints.zsChildLock:
                return {child_lock: value ? 'LOCK' : 'UNLOCK'};
            case tuyaLocal.dataPoints.zsFaulty:
                meta.logger.debug('converter TS0601 => zsFaulty : dp=' + dp + ' data=' + JSON.stringify(msg.data));
                meta.logger.debug('converter TS0601 => zsFaulty : result value=' + value[0]);
                return {faulty: value[0]};
            case tuyaLocal.dataPoints.zsHolidaySet:
                meta.logger.info('converter TS0601 => zsHolidaySet : dp=' + dp + ' data=' + JSON.stringify(msg.data));
                meta.logger.info('converter TS0601 => zsHolidaySet : result value=' + value[0]);
                return {holiday_set: value[0]};
            case tuyaLocal.dataPoints.zsKSSW:
                meta.logger.debug('converter TS0601 zsKSSW : dp=' + dp + ' data=' + JSON.stringify(msg.data));
                return {zsKSSW_delay_in_second: value};
            case tuyaLocal.dataPoints.zsKCWD:
                meta.logger.debug('converter TS0601 zsKCWD : dp=' + dp + ' data=' + JSON.stringify(msg.data));
                return {zsKCWD_open_window_temp_chg: (value / 10).toFixed(1)};
            case tuyaLocal.dataPoints.zsSSWD:
                meta.logger.debug('converter TS0601 zsSSWD : dp=' + dp + ' data=' + JSON.stringify(msg.data));
                return {zsSSWD_confort_temp: (value / 10).toFixed(1)};
            case tuyaLocal.dataPoints.zsJLWD:
                meta.logger.debug('converter TS0601 zsJLWD : dp=' + dp + ' data=' + JSON.stringify(msg.data));
                return {zsJLWD_eco_temp: (value / 10).toFixed(1)};
            case tuyaLocal.dataPoints.zsHeatingStopping:
                meta.logger.debug('converter TS0601 => zsHeatingStopping : dp=' + dp + ' data=' + JSON.stringify(msg.data));
                meta.logger.debug('converter TS0601 => zsHeatingStopping : result value=' + JSON.stringify(msg.data));
                return {heating_stopping: tuyaLocal.thermostatOnOff[value]};
            case tuyaLocal.dataPoints.zsProgramMonday:
                meta.logger.debug('converter TS0601 zsProgramMonday : dp=' + dp + ' data=' + JSON.stringify(msg.data));
                return {zsProgramMonday: 'not_implemented'};
            case tuyaLocal.dataPoints.zsProgramWednesday:
                meta.logger.debug('converter TS0601 zsProgramWednesday : dp=' + dp + ' data=' + JSON.stringify(msg.data));
                return {zsProgramWednesday: 'not_implemented'};
            case tuyaLocal.dataPoints.zsProgramFriday:
                meta.logger.debug('converter TS0601 zsProgramFriday : dp=' + dp + ' data=' + JSON.stringify(msg.data));
                return {zsProgramFriday: 'not_implemented'};
            case tuyaLocal.dataPoints.zsProgramSunday:
                meta.logger.debug('converter TS0601 zsProgramSunday : dp=' + dp + ' data=' + JSON.stringify(msg.data));
                return {zsProgramSunday: 'not_implemented'};
            case tuyaLocal.dataPoints.zsProgramTuesday:
                meta.logger.debug('converter TS0601 zsProgramTuesday : dp=' + dp + ' data=' + JSON.stringify(msg.data));
                return {zsProgramTuesday: 'not_implemented'};
            case tuyaLocal.dataPoints.zsProgramThursday:
                meta.logger.debug('converter TS0601 zsProgramThursday : dp=' + dp + ' data=' + JSON.stringify(msg.data));
                return {zsProgramThursday: 'not_implemented'};
            case tuyaLocal.dataPoints.zsProgramSaturday:
                meta.logger.debug('converter TS0601 zsProgramSaturday : dp=' + dp + ' data=' + JSON.stringify(msg.data));
                return {zsProgramSaturday: 'not_implemented'};
            case tuyaLocal.dataPoints.zsOnline:
                meta.logger.debug('converter TS0601 => zsOnline : dp=' + dp + ' data=' + JSON.stringify(msg.data));
                meta.logger.debug('converter TS0601 => zsOnline : result value=' + tuyaLocal.thermostatOnlineStatus[value]);
                return {online_state: tuyaLocal.thermostatOnlineStatus[value]};
            default:
                meta.logger.warn(`zigbee-herdsman-converters:zsThermostat: Unrecognized DP #${dp} with data ${JSON.stringify(msg.data)}`);
            }
            return {dp_no_implemented : dp};
        },
    },
};
const tzLocal = {
    zs_thermostat_child_lock: {
        key: ['child_lock'],
        convertSet: async (entity, key, value, meta) => {
            await tuya.sendDataPointBool(entity, tuyaLocal.dataPoints.zsChildLock, value === 'LOCK');
        },
    },
    zs_thermostat_current_heating_setpoint: {
        key: ['current_heating_setpoint'],
        convertSet: async (entity, key, value, meta) => {
            await tuya.sendDataPointValue(entity, tuyaLocal.dataPoints.zsHeatingSetpoint, Math.round(value * 10));
        },
    },
    zs_thermostat_preset_mode: {
        key: ['preset'],
        convertSet: async (entity, key, value, meta) => {
            const lookup = {'schedule': 0, 'manual': 1, 'holiday': 3};
            await tuya.sendDataPointEnum(entity, tuyaLocal.dataPoints.zsMode, lookup[value]);
            if (value == 'manual') {
                const temp = globalStore.getValue(entity, 'current_heating_setpoint');
                if (temp) {
                    await tuya.sendDataPointValue(entity, tuyaLocal.dataPoints.zsHeatingSetpoint, Math.round(temp * 10));
                }
            }
        },
    },
};
const device = {
    // Tuya Radiator Valve
    zigbeeModel: ['TS601'],
    fingerprint: [{modelID: 'TS0601', manufacturerName: '_TZE200_hue3yfsn'}],
    model: 'TV02-GV',
    vendor: 'SILUMEN',
    description: 'Radiator valve with thermostat',
    fromZigbee: [
        fz.tuya_data_point_dump,
        fz.ignore_basic_report,
        fz.ignore_tuya_set_time,
        fzLocal.zs_thermostat,
    ],
    toZigbee: [
        tz.tuya_data_point_test,
        tzLocal.zs_thermostat_current_heating_setpoint,
        tzLocal.zs_thermostat_child_lock,
        tzLocal.zs_thermostat_preset_mode,
    ],
    onEvent: tuya.onEventSetLocalTime,
    meta: {},
    configure: async (device, coordinatorEndpoint, logger) => {
        const endpoint = device.getEndpoint(1);
        await reporting.bind(endpoint, coordinatorEndpoint, ['genBasic']);
    },
    exposes: [
        exposes.climate().withSetpoint('current_heating_setpoint', 0.5, 30.0, 0.5)
                         .withLocalTemperature()
                         .withLocalTemperatureCalibration()
                         .withPreset(['schedule', 'manual', 'holiday']),
        e.battery(),
        e.child_lock(),
        e.week(),
        e.window_detection(),
        exposes.enum('online_state', exposes.access.STATE, ["online","offline"]).withDescription('Online State'),
            ],
};

module.exports = device;
```

**Comment by Oddiwahn** [comment]
Hi, thanks for your effort guys! I received my TRVs today and already checked your code, it is working nicely!

I have some comments:

- got some errors from fz.tuya_data_point_dump, and tz.tuya_data_point_test,
- you should add ea.STATE_SET when you can read AND write or ea.STATE when read only in the exposes
- your datapoint list is correct, I checked on iot.tuya.com

Also, I have some thoughts as I am currently developing a converter too (see #3190). 
In the exposes of the "climate" entity, there are three different types of states/modes:

Nr  | name | direction | possible options | 
-----|--------------------|----------|-----------------
3 | SystemMode | get & set | ['off', 'heat', 'cool', 'auto', 'dry', 'fan_only', 'sleep', 'emergency_heating']
1 | RunningState | get  | ['idle', 'heat', 'cool', 'fan_only']
2 |Preset | get & set | ANY

![image](https://user-images.githubusercontent.com/91994384/140419791-24d47e74-ed9c-459f-83cf-e83c60b3ec4f.png)

Now the thing is, that if you are using the Climate panel in home assistant, only what is specified in "SystemMode" can be controlled by the buttons on the bottom (nr 3 in picture). The four modes in this case are "Auto", "Manual", "Holiday" and "Boost". 
Holiday appears to be something like "off" (data point 107 heating stop) until the date specified with datapoint 46 (holidaySet). Both datapoints can be set and get.
Boost is basically "heat", (dp 115, can be set and get) for a predefined time whose current value can be read from data point 101 (get only, boosttime)
If you do not specify "systemmode", all of the buttons will appear as in the picture. So in my converter I used 'off', 'heat',  'auto' and 'fan_only' (fan_only = manual). That way I have only four buttons which have a function. My other TRVs are also exposing a workings state (read only) for what I use "runningState". In your case this doen't apply here. 
On the other hand you have the comfort and eco temperatures. I would use "preset" to set the temperature to the "eco temperature" dp 104 or "comfort temperature" dp 105 values. This is just what I'm thinking which I#d like to share. Maybe you have some different thoughts which I'd be interested in.

Finally, I am struggeling with setting the "local_temperature_calibration". This would actually be nice to have. The following code is what most converters use but it seems neither to work for my Rti-Tek TRVs nor for this one. Also, I don't understand the "0xFFFFFFFF + temp + 1;" part.

    zs_thermostat_temperature_calibration: {
        key: ['local_temperature_calibration'],
        convertSet: async (entity, key, value, meta) => {
            var temp = Math.round(value * 10);
            if (temp < 0) {
                temp = 0xFFFFFFFF + temp + 1;
            }
            await tuya.sendDataPointValue(entity, tuyaLocal.zsCorrectionTemp, temp);
        },
    },

Hope this helps! 
Chris

**Comment by handfreezer** [comment]
@Oddiwahn : may i ask how to find your specific dp on iot.tuya.com?

**Comment by Oddiwahn** [comment]
> @Oddiwahn : may i ask how to find your specific dp on iot.tuya.com?

Sure ;-)
https://www.zigbee2mqtt.io/advanced/support-new-devices/03_find_tuya_data_points.html#requirements-and-caveats

**Comment by handfreezer** [comment]
OK, I missed this, what a shame... So, I'll have to buy a tuya gateway ...

**Comment by handfreezer** [comment]
I feel having a problem : local_temperature is updated while pairing the valve _TZE200_hue3yfsn , but after the pairing process, it seems that there is no more update of local_temperature .
Do you think it is possible?

**Comment by vladi1234** [comment]
Hallo zusammen, 
Warum muss ein konverter einstellungen für HA haben?

Zur info, Ich habe Heute einen Pull Request für Tuya TV02-Zigbee _TZE200_hue3yfsn schon erstellt.

**Comment by Oddiwahn** [comment]
> Hello, everyone, Why does a converter have to have settings for HA?
> 
> For info, I have already created a pull request for TV02-Zigbee _TZE200_hue3yfsn today. I think it won't be long before it is supported.

Did you test local temperature calibration? Is it working? I'm really keen on getting this running

**Comment by handfreezer** [comment]
@vladi1234 : I tested your pull request and made you few returns on it.

**Comment by insipiens** [comment]
@handfreezer  surely you could just have an automation that sends a calbration message if you want the local temp updated.

I will admit I’m happy that it preserves battery by not transmitting this regularly. But understand that might be a minority viewpoint 😬

**Comment by handfreezer** [comment]
Yes, in fact, I need to get this update to avoid installation of a specific sensor for temp.
The main fact is that I failed to find a command to activate reporting regularly of local_temp . This is only point making this TRV not a valuable one for my installation.
I'll try to connect through iot.tuya.com to see how local_temp is updated, if I didn't find, i'll give up this TRV.

**Comment by vladi1234** [comment]
@kalmatthew I like to discuss all questions, but for issues that is here and not for Pull Request. Please try to keep Pull Request clean. And please bring this https://github.com/Koenkk/zigbee-herdsman-converters/pull/3291#issuecomment-962011088 here in this 3142 isues.

**Comment by vladi1234** [comment]
There are already so many similar thermostats that work, but different. That's why they make too many mistakes.
I try as best as possible to integrate our TV02_Thermostat into z2m. Unfortunately it is not completely finished, as @handfreezer has already noticed.

**Comment by vladi1234** [comment]
> @handfreezer  : Also, I do not understand the holidaye mode date format:
> ![image](https://user-images.githubusercontent.com/45510877/140576118-f53d2852-7e38-4a08-8bd5-0119c3a1fb83.png)

Holiday mode can only be switched on if you set the start and end date and time.
"data": "202101010101202101010101"

```
              Data        Time
Start:     2021.01.01    01:01
End:       2021.01.01    01:01
```
The only thing missing is the right converter. But I'm working on it.

**Comment by vladi1234** [comment]
> @handfreezer : Refreshing manually local_temperature is giving an error:
> ![image](https://user-images.githubusercontent.com/45510877/140578406-b956a72f-fe4b-47d0-9672-20d8bb891130.png)

Unfortunately there is only exposes local_temperature with Refreshing_Button, which shouldn't be there because this function does not exist and local temperature must not be queried. TRV sends local temperature only when TRV deems it necessary.

**Comment by vladi1234** [comment]
> > Hello, everyone, Why does a converter have to have settings for HA?
> > For info, I have already created a pull request for TV02-Zigbee _TZE200_hue3yfsn today. I think it won't be long before it is supported.
> 
> Did you test local temperature calibration? Is it working? I'm really keen on getting this running

Yes, is worked.

**Comment by handfreezer** [comment]
> > @handfreezer  : Also, I do not understand the holidaye mode date format:
> > ![image](https://user-images.githubusercontent.com/45510877/140576118-f53d2852-7e38-4a08-8bd5-0119c3a1fb83.png)
> 
> Holiday mode can only be switched on if you set the start and end date and time. "data": "202101010101202101010101"
> 
> ```
>               Data        Time
> Start:     2021.01.01    01:01
> End:       2021.01.01    01:01
> ```
> 
> The only thing missing is the right converter. But I'm working on it.

OK, but, does this mean that the TRV will have to be set with a timestamp?

**Comment by handfreezer** [comment]
> > @handfreezer : Refreshing manually local_temperature is giving an error:
> > ![image](https://user-images.githubusercontent.com/45510877/140578406-b956a72f-fe4b-47d0-9672-20d8bb891130.png)
> 
> Unfortunately there is only exposes local_temperature with Refreshing_Button, which shouldn't be there because this function does not exist and local temperature must not be queried. TRV sends local temperature only when TRV deems it necessary.

OK, but it seems that withLocalTemperature can take an "access" parameter set by default to 'STATE_GET', but you can change it to STATE, like:
```
withLocalTemperature(exposes.access.STATE)
```

**Comment by vladi1234** [comment]
> @handfreezer I like to discuss all questions, but for issues that is here and not for Pull Request. Please try to keep Pull Request clean. And please bring this [#3291 (comment)](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3291#issuecomment-962011088) here in this 3142 isues.

@handfreezer Have you already done what I asked you to do?

**Comment by handfreezer** [comment]
Just to tell you that : 
the week planning conversion is strange:
![image](https://user-images.githubusercontent.com/2912752/140539538-6f657b48-c405-43eb-bf35-919187bcd273.png)

Also, I do not understand the holidaye mode date format:
![image](https://user-images.githubusercontent.com/2912752/140539607-739a0d11-48c0-48e1-806e-5d96370d357b.png)

Refreshing manually local_temperature is giving an error:
![image](https://user-images.githubusercontent.com/2912752/140539719-f7400822-9949-4e76-b914-ff58511217e2.png)

Also, is it possible to get on top le current_heating_setpoint, local_temperature and preset mode ?

> > @handfreezer I like to discuss all questions, but for issues that is here and not for Pull Request. Please try to keep Pull Request clean. And please bring this [#3291 (comment)](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3291#issuecomment-962011088) here in this 3142 isues.
> 
> @handfreezer Have you already done what I asked you to do?

@vladi1234 : sorry, but as you started to answer me in this issue, I didn't feel necessary to copy it.

**Comment by vladi1234** [comment]
@handfreezer Can you possibly delete the comment for Pull Request 3142, or change it so that there remains a small point?

Thanks!

**Comment by vladi1234** [comment]
> > > @handfreezer  : Also, I do not understand the holidaye mode date format:
> > > ![image](https://user-images.githubusercontent.com/45510877/140576118-f53d2852-7e38-4a08-8bd5-0119c3a1fb83.png)
> > 
> > 
> > Holiday mode can only be switched on if you set the start and end date and time. "data": "202101010101202101010101"
> > ```
> >               Data        Time
> > Start:     2021.01.01    01:01
> > End:       2021.01.01    01:01
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > The only thing missing is the right converter. But I'm working on it.
> 
> OK, but, does this mean that the TRV will have to be set with a timestamp?

Manual:
8.3 Holiday Mode
When you go out or go to a party, you will use the holiday mode. **The holiday mode will automatically start at the set time starting point and run the holiday temperature.**
**When the holiday mode is not activated, and you want to view the holiday mode settings, please press the pair button to view, the holiday mode icon ( ⛱ ) will flash for 5 seconds and then it will automatically return to the auto mode.** If you want to end your holiday mode earlier, press the pair button to cancel.
App operation: Click the Mode icon in the App control panel, and switch to any other mode to cancel holiday mode.

**Comment by handfreezer** [comment]
As planned, I change log_level to debug and joined my device as thermostat_1 (changing fingerprint), you will see there is a lot a warning telling me it is not the right way to integrate it.
[hue_as_thermostat_1.txt](https://github.com/Koenkk/zigbee-herdsman-converters/files/7490767/hue_as_thermostat_1.txt)

**Comment by handfreezer** [comment]
> @handfreezer Can you possibly delete the comment for Pull Request 3142, or change it so that there remains a small point?
> 
> Thanks!

For sure, done!

**Comment by vladi1234** [comment]
> > > @handfreezer : Refreshing manually local_temperature is giving an error:
> > > ![image](https://user-images.githubusercontent.com/45510877/140578406-b956a72f-fe4b-47d0-9672-20d8bb891130.png)
> > 
> > 
> > Unfortunately there is only exposes local_temperature with Refreshing_Button, which shouldn't be there because this function does not exist and local temperature must not be queried. TRV sends local temperature only when TRV deems it necessary.
> 
> OK, but it seems that withLocalTemperature can take an "access" parameter set by default to 'STATE_GET', but you can change it to STATE, like:
> 
> ```
> withLocalTemperature(exposes.access.STATE)
> ```
> 
> Also, do you know if there is a plan to activate or force the TRV to send update, as mine is running from know more that 24hours with variation local temp and no update :(



Yes, you can change 'STATE_GET' ----> to ----> 'STATE', you are right, but it then leads to a global conflict with other thermostats that have this function.

**Comment by vladi1234** [comment]
@handfreezer I solved the problem with button in a different way.

![image](https://user-images.githubusercontent.com/45510877/140647826-5f47cef5-2861-4859-818f-53e289c76a0e.png)

**Comment by vladi1234** [comment]
TV02-Zigbee _TZE200_hue3yfsn is marged!

**Comment by rforro** [comment]
Does anybody has an idea in which format has to be the data for schedule? I'm trying to set scheduler for Monday, but the device won't confirm/send the parameters back.

I understand the data format `[36, 0, 170, 72, 0, 210, 84, 0, 170, 102, 0, 210, 144, 0, 170, 144, 0, 210, 144, 0, 170, 144, 0, 210, 144, 0, 170, 144, 0, 210]` but when I'm trying to send the same array with just a minor change in one temperature parameter, the TRV ignores me. Does anybody has an idea?

**Comment by vladi1234** [comment]
```
36, 0, 170,                        
36/6=6, 0=0, 170/10=17,       
06:00   17°C                      

72, 0, 210,
72/6=12, 0=0, 210/10=21
12:00   21°C
```

![image](https://user-images.githubusercontent.com/45510877/141656992-c555d1c7-734f-411f-a254-20d41de1bf89.png)

**Comment by handfreezer** [comment]
Hello,  to test the merge of vlad push, I change in Z2M package.json and npm-shrink.json the converters dependencies from .303 to .317, then did a 'npm ci', but I failed to update the dependency. I guess it is based on me because I don't know how to do it, and it seems not simple as changing the version in those JSON file.
Can you help please?

**Comment by vladi1234** [comment]
Simply do a complete z2m update.

**Comment by vladi1234** [comment]
sudo systemctl stop zigbee2mqtt && cd /opt/zigbee2mqtt && git checkout HEAD -- npm-shrinkwrap.json && git fetch && git fetch && git checkout dev && git pull && npm ci && cd &&  sudo systemctl start zigbee2mqtt

**Comment by handfreezer** [comment]
Hello,
I feel like this is what I tried few days ago, but I still stuck to a zigbee-converter oldest than the merge of your push.
I'll try again the weekend.

**Comment by RaW79** [comment]
Hello,
also trying to use _TZE200_hue3yfsn and guessing that it is based on your marged master by showing the device name TV02-Zigbee after pairing. May I kindly ask for further information how an update of local temperature is implemented according @vladi1234 note before? I tried by manual update of the temperature calibration but no update of local temperature works for me. Only by pairing this device it is updated.
In Addition, do you have any information regarding valve position? On z2m device support list a DP "position" is listed but i cannot find or get it from the merged master converter?

Many thx!

**Comment by vladi1234** [comment]
Hi, 
I guess you looked at the wrong Thermastet. I wrote earlier that these two thermostats are completely different. Our TV02-Zigbee _TZE200_hue3yfsn is not yet in the z2m supported list.

>
 ```
// TV02-Zigbee Thermostat
tvMode: 2,
tvWindowDetection: 8,
tvFrostDetection: 10,
tvHeatingSetpoint: 16,
tvLocalTemp: 24,
tvTempCalibration: 27,
tvWorkingDay: 31,
tvHolidayTemp: 32,
tvBattery: 35,
tvChildLock: 40,
tvErrorStatus: 45,
tvHolidayMode: 46,
tvBoostTime: 101,
tvOpenWindowTemp: 102,
tvComfortTemp: 104,
tvEcoTemp: 105,
tvWeekSchedule: 106,
tvHeatingStop: 107,
tvMondaySchedule: 108,
tvWednesdaySchedule: 109,
tvFridaySchedule: 110,
tvSundaySchedule: 111,
tvTuesdaySchedule: 112,
tvThursdaySchedule: 113,
tvSaturdaySchedule: 114,
tvBoostMode: 115,

// Haozee TS0601 TRV       thermostat_1.txt
haozeeSystemMode: 1,
haozeeHeatingSetpoint: 2,
haozeeLocalTemp: 3,
haozeeBoostHeating: 4,
haozeeBoostHeatingCountdown: 5,
haozeeRunningState: 6,
haozeeWindowState: 7,
haozeeWindowDetection: 8,
haozeeChildLock: 12,
haozeeBattery: 13,
haozeeFaultAlarm: 14,
haozeeMinTemp: 15,
haozeeMaxTemp: 16,
haozeeScheduleMonday: 17,
haozeeScheduleTuesday: 18,
haozeeScheduleWednesday: 19,
haozeeScheduleThursday: 20,
haozeeScheduleFriday: 21,
haozeeScheduleSaturday: 22,
haozeeScheduleSunday: 23,
haozeeTempCalibration: 101,
haozeeValvePosition: 102,
haozeeSoftVersion: 150,
```
**If you look closely at DP, they're totally different. That is, thermostats are totally different.
Nue and thermostat_1 cannot be compared!**

**Comment by RaW79** [comment]
@vladi1234 thx for you feedback and fully understand difference of those 2 TV models and so they are not comparable.

Nevertheless I am trying to use TV02-Zigbee _TZE200_hue3yfs and as it was discussed the DP
`tvLocalTemp: 24,`
is not updated automatically. You mentioned that this is solved "I solved the problem with button in a different way."

If I am right the merged master is not only used by z2m but also e.g. by iobroker for implementation. 
So assuming that you solution might not work or is disabled in other usage I like to check/understand If I can build it manually and what to do. Update of DP:
`tvTempCalibration: 27,`
does not work for me. If you have made usage of this DP for update of 24 I cannot do anything by myself....

**Comment by vladi1234** [comment]
> Nevertheless I am trying to use TV02-Zigbee _TZE200_hue3yfs and as it was discussed the DP
tvLocalTemp: 24,
is not updated automatically. You mentioned that this is solved "I solved the problem with button in a different way."

If you read carefully. It was about a **button** that shouldn't be in the front, but wasn't about the automatic update function tvLocalTemp: 24 ,.

**Comment by RaW79** [comment]
@vladi1234 
fully understand and agree on that. Just asked for the function behind what is used to update tvLocalTemp: 24 or just in general understand what is done.

Based on the information available I assumed that updating tvTempCalibration: 27 is the common method used for this.
So I manually changed the DP within iobroker but the tvLocalTemp: 24 isn´t changed/updated.
Because I also do not get a debug message by updating/changing tvTempCalibration: 27 this should be a iobroker specific topic, not to discuss here.

But in general I´m still interested to get to know if this is still the point of interest to look for if tvLocalTemp: 24 is requested to be updated within automatic procedure.

Thx

**Comment by handfreezer** [comment]
> Hello, I feel like this is what I tried few days ago, but I still stuck to a zigbee-converter oldest than the merge of your push. I'll try again the weekend.

OK, I missed the git checkout dev ... And now I have a few notes for you ;-)

**Comment by handfreezer** [comment]
Testing dev branch, I have this when trying to change holiday value:

```
Zigbee2MQTT:debug 2021-11-22: Received MQTT message on 'zigbee2mqtt/0x2c1165fffe7c2f38/set' with data '{"holiday_mode_date":2.021010101012021e+23}'
Zigbee2MQTT:debug 2021-11-22: Publishing 'set' 'holiday_mode_date' to '0x2c1165fffe7c2f38'
Zigbee2MQTT:error 2021-11-22: Publish 'set' 'holiday_mode_date' to '0x2c1165fffe7c2f38' failed: 'TypeError: Command 0x2c1165fffe7c2f38/1 manuSpecificTuya.setData({"status":0,"transid":13,"datatype":5,"length_hi":0,"length_lo":0,"data":2.021010101012021e+23}, {"sendWhenActive":false,"timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":true,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (values is not iterable)'
Zigbee2MQTT:debug 2021-11-22: TypeError: Command 0x2c1165fffe7c2f38/1 manuSpecificTuya.setData({"status":0,"transid":13,"datatype":5,"length_hi":0,"length_lo":0,"data":2.021010101012021e+23}, {"sendWhenActive":false,"timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":true,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (values is not iterable)
    at BuffaloZcl.writeListUInt8 (/home/ether/z2m/zigbee2mqtt.clone/node_modules/zigbee-herdsman/src/buffalo/buffalo.ts:168:29)
    at BuffaloZcl.write (/home/ether/z2m/zigbee2mqtt.clone/node_modules/zigbee-herdsman/src/buffalo/buffalo.ts:266:18)
    at BuffaloZcl.write (/home/ether/z2m/zigbee2mqtt.clone/node_modules/zigbee-herdsman/src/zcl/buffaloZcl.ts:376:26)
    at ZclFrame.writePayloadCluster (/home/ether/z2m/zigbee2mqtt.clone/node_modules/zigbee-herdsman/src/zcl/zclFrame.ts:158:21)
    at ZclFrame.toBuffer (/home/ether/z2m/zigbee2mqtt.clone/node_modules/zigbee-herdsman/src/zcl/zclFrame.ts:78:18)
    at ZStackAdapter.sendZclFrameToEndpointInternal (/home/ether/z2m/zigbee2mqtt.clone/node_modules/zigbee-herdsman/src/adapter/z-stack/adapter/zStackAdapter.ts:388:22)
    at Object.func (/home/ether/z2m/zigbee2mqtt.clone/node_modules/zigbee-herdsman/src/adapter/z-stack/adapter/zStackAdapter.ts:356:25)
    at Queue.executeNext (/home/ether/z2m/zigbee2mqtt.clone/node_modules/zigbee-herdsman/src/utils/queue.ts:32:42)
    at /home/ether/z2m/zigbee2mqtt.clone/node_modules/zigbee-herdsman/src/utils/queue.ts:21:18
    at new Promise (<anonymous>)

```

**Comment by handfreezer** [comment]
I feel like window_detection should not be clickable, as it is changing state of the TRV, i do not feel it is an ON/OFF of the function.

**Comment by handfreezer** [comment]
Small icon is missing

**Comment by handfreezer** [comment]
About updating the local_temperature, I understood that a change on local_temperature_calibration implies that the TRV is sending an update of local_temperature. Is it possible to set a timer to change that for an automatic way in Z2M?

**Comment by insipiens** [comment]
Perhaps not everyone wants that @handfreezer . Can’t you just automate the calibration in your host software? Much easier and offers more flexibility e.g less frequent checks overnight or when away.

making the device do what it is not designed to do will be a win/lose situation.

or you could just sell this on ebay and buy a device that reports local temp regularly.

**Comment by vladi1234** [comment]
What holiday_mode_date and window_detection is also known that does not work correctly yet, it takes time. But thank you anyway for commenting and testing.

Yes, that's correct, locale_temp update works if you change local_temp_correction after a certain time. But I think that's the wrong way, there are other ways to query local_temp. I need time to test all of this.

But in my opinion, the manufacturer certainly did not plan to use local_temp as a thermosensor!
Normally the termosensor has to be located in the middle of the room and has to measure the room temperature!
Local_temp was planned by the manufacturer for internal control in the thermostat and it also does not show the correct room temperature!

**Comment by vladi1234** [comment]
> Small icon is missing

@handfreezer What do you mean by that?

**Comment by RaW79** [comment]
> I don't use iobroker so I can't comment on it.
> 
> > Update of DP:
> > tvTempCalibration: 27,
> > does not work for me.
> 
> I want to know what exactly is not working?

Small change but big difference :). Temperature calibration and each of the other DPs which can be set manually have to be data type string not number. Now it works according your description also for me, but of course looking forward if you can improve further.

Thanks and best regards.

**Comment by vladi1234** [comment]
Today I tested all possible variants of lacal_temp queries and found that the only one and the best solution is to send local_temperature_calibration = 0, then TVR delivers all new data including local_temp.

You can set any time interval. Everyone can adjust the individual for themselves. You can do that with every home automation.

But do not overdo it with interval, battery consumption can increase!

It works great!

I created an example Domoticz:


![image](https://user-images.githubusercontent.com/45510877/143069726-74957626-41a4-45df-aadc-572e984aef65.png)

**Comment by vladi1234** [comment]
New problem auto_mode starts when you press hs = off or fs = off. Own logick in TRV.
That makes problems when writing scenarios for TRV.

**Comment by vladi1234** [comment]
Deverses fix and open_window function #3419

**Comment by ypoosn** [comment]
Hi,

did anybody already contacted the manufacturer of the device (Zonnsmart TV02-ZG)
[https://expo.tuya.com/product/947044](https://expo.tuya.com/product/947044) to get help activating a periodic
or change based update of the current temperature from the device itself (secret tuya commands?)

Currently I am using the workaround polling each 15 min.

They could sell even more devices (TV01-ZG, TV02-ZG, TV05-ZG) if z2m compatibility 
increases  and each type is z2m approved ;-)

Thanks for your great efforts in making this device work with z2m.

best, Ypo

**Comment by vladi1234** [comment]
I added ` _TZE200_e9ba97vf, _TZE200_husqqvux` to my `_TZE200_hue3yfsn` code today.
Please test whether everything is OK.

Merged #3469 into master.

**Comment by rforro** [comment]
I bought the Tuya Gateway to observe how this TRV behaves with it and immediatelly discovered, that when I opened the TRV in the Tuya App the antenna icon on the TRV lights up. This causes the periodical publishing of temperature, what are we looking for. After closing the app the icon is gone.

**Comment by insipiens** [comment]
Which means it interrogates it only during an app session. The ideal here would be a switch, or similar to allow replication of that behaviour.

Time to dust off my zigbee sniffer?

**Comment by rforro** [comment]
Here is log from Tuya Iot Platform where I
* set childlock through the app
* turned it off,
* let tablet to turn itself off, which ended the "app session" and antenna symbol on TRV lit off.

2021-12-10 20:55:24 | Report | Check01 | on | device itself |  
2021-12-10 20:54:21 | Report | Thursday program | MACqSADckACqkADSkACqkADSkACqkADSkACqkADS | device itself |  
2021-12-10 20:54:21 | Report | Wednesday program | MACqSADckACqkADSkACqkADSkACqkADSkACqkADS | device itself |  
2021-12-10 20:54:21 | Report | Tuesday program | MACqSADckACqkADSkACqkADSkACqkADSkACqkADS | device itself |  
2021-12-10 20:54:21 | Report | Monday program | MACqSADckACqkADSkACqkADSkACqkADSkACqkADS | device itself |  
2021-12-10 20:54:21 | Report | Sunday program | MACqSADckACqkADSkACqkADSkACqkADSkACqkADS | device itself |  
2021-12-10 20:54:21 | Report | Saturday program | MACqSADckACqkADSkACqkADSkACqkADSkACqkADS | device itself |  
2021-12-10 20:54:21 | Report | Friday program | MACqSADckACqkADSkACqkADSkACqkADSkACqkADS | device itself |  
2021-12-10 20:54:20 | Report | Battery Capacity | 100% | device itself |  
2021-12-10 20:54:19 | Report | Battery Capacity | 0% | device itself
2021-12-10 20:54:18 | Report | Check01 | off | device itself
2021-12-10 20:53:28 | Report | Lock KEY | OFF | device itself
2021-12-10 20:53:15 | Report | Thursday program | MACqSADckACqkADSkACqkADSkACqkADSkACqkADS | device itself |  
2021-12-10 20:53:15 | Report | Wednesday program | MACqSADckACqkADSkACqkADSkACqkADSkACqkADS | device itself |  
2021-12-10 20:53:15 | Report | Tuesday program | MACqSADckACqkADSkACqkADSkACqkADSkACqkADS | device itself |  
2021-12-10 20:53:14 | Report | Monday program | MACqSADckACqkADSkACqkADSkACqkADSkACqkADS | device itself |  
2021-12-10 20:53:14 | Report | Sunday program | MACqSADckACqkADSkACqkADSkACqkADSkACqkADS | device itself |  
2021-12-10 20:53:14 | Report | Saturday program | MACqSADckACqkADSkACqkADSkACqkADSkACqkADS | device itself |  
2021-12-10 20:53:14 | Report | Friday program | MACqSADckACqkADSkACqkADSkACqkADSkACqkADS | device itself |  
2021-12-10 20:53:13 | Report | Battery Capacity | 100% | device itself |  
2021-12-10 20:53:13 | Report | Battery Capacity | 0% | device itself
2021-12-10 20:53:12 | Report | Check01 | on | device itself |  
2021-12-10 20:53:09 | Report | Lock KEY | ON | device itself

As you can see the "Check01" DP which is 115 has something to do with the "app session". It will be great if you could sniff the packets, because I've never done it and currently don't have the time learn it.

**Comment by vladi1234** [comment]
@rforro If you are in the APP, how long does the antenna light up or as long as you are in the APP?
In which case does the antenna go out?

**Comment by rforro** [comment]
@vladi1234 The antenna icon first light up when I open the TRV device in the app, not on list of all devices. As long as the device stays opened in the app the antenna icon is on. It turns off in following three cases:
* closing the device in the app / going back to all devices list
* closing the app
* turning off the tablet/mobile

in all three cases it happens after 30-60sec, it took never longer than 60 seconds but one time it happened after 20 seconds. Probably the app has to "ping" the device every now and then to stay in "online" mode.

**Comment by vladi1234** [comment]
Do I understand correctly that to change the settings for TVR in the APP, it must always be online / on?

**Comment by rforro** [comment]
You cannot do it otherway. When you opens the TRV in the app, it connects (aka the antenna icon on the TRV lights up) automatically.

**Comment by mgwoj** [comment]
Why it is not possible to edit weekly schedule from the UI of z2m?
I see in the code that scheduling options are defined as `ea.STATE` instead of `ea.STATE_SET` - `.withFeature(exposes.text('schedule_monday', ea.STATE))`. What is the reason behind that?

**Comment by insipiens** [comment]
> Why it is not possible to edit weekly schedule from the UI of z2m?
> I see in the code that scheduling options are defined as `ea.STATE` instead of `ea.STATE_SET` - `.withFeature(exposes.text('schedule_monday', ea.STATE))`. What is the reason behind that?

PR ftw!

**Comment by vladi1234** [comment]
As for the automatic schedule, I'm still working on it.

Just change it to ea.STATE_SET, it won't work anyway.

But there's a bigger problem that TVR doesn't accept converter commands.

I can't force TVR to accept komand yet.

Maybe it has something to do with the online function, I'll test whether it is related.

Maybe that won't be feasible, I don't know.

**But in our case, however, no schedule is required as we all use home automation programs and only manual mode is required to control the TVR, which is logical!**

**Comment by jeykodev** [comment]
@vladi1234 thank you so much for you work on this TRV. Can you already say roughly whether there will be a "real" solution for the local_temp problem in the future? Or will the community be forced to use the workaround (setting temp calibration on a regular base).

BTW: I have the feeling that Aliexpress (and other stores) are now flooded with the new hue3yfsn type of TS0601 thermostat: 

![image](https://user-images.githubusercontent.com/10532676/147294898-59d2f40e-aa1c-433b-b0ca-56c7c43d721a.png)

I feel that they gonna replace the common used old TS0601:

![image](https://user-images.githubusercontent.com/10532676/147294941-f8508937-390d-48ef-a5f4-8c5c98777465.png)

**Comment by stoinov** [comment]
I have both and here are my findings for their UX:
- The new one with rotary knob is better for setting temp manually as the touch sensitivity on the old one was all over the place.
- The old one has the advantage of rotating the temp display if you happen to place it upside down. (Not sure if the new one has this ability but I haven't found a reference for it).
- The old one has the advantage of showing and setting the valve position in %, which in some cases is better preferred than just using temp.
- The UI on the new one is cleaner and more readable than the old one which felt chaotic at times.
- The new one has metal ring while the old had a plastic one. All additional adapters are plastic for both.

**Comment by jeykodev** [comment]
> But in our case, however, no schedule is required as we all use home automation programs and only manual mode is required to control the TVR, which is logical!

No, I don't agree with that! Setting up the schedule on the valve directly can be very handy. For example: when you are using window sensors you can setup a workflow very easy: window open -> preset manual (0°C) / window closed -> preset auto. So when you are closing the window, the thermostat is back into its schedule. Speaking for HA this is not an easy task to todo without the schedule on the valve. 

For me personally it's a must have feature. Let me know if I can help with the implementation.

**Comment by aso824** [comment]
It would be great to get valve position to control central heating furnace. Any chance for it?
Does this TRV shows valve position in official app?

**Comment by vladi1234** [comment]
> It would be great to get valve position to control central heating furnace. Any chance for it? Does this TRV shows valve position in official app?

As far as I know, this function does not exist for him.

**Comment by vladi1234** [comment]
I made PR for **holiday_mode** yesterday, and now I want to make screenshot with new interface.

![image](https://user-images.githubusercontent.com/45510877/151700767-a8ff3ee3-3d3e-45ac-a6bf-fc989c919710.png)

**Comment by jeykodev** [comment]
@vladi1234 thank you! Are you planing on work on the schedule feature too?

**Comment by vladi1234** [comment]
I'm also trying to crack Schedule, but Schedule has a complex protocol unknown to me. I also don't have a Tuya Gateway to listen to.

**Comment by neil1111** [comment]
Hi. I purchased three of these models on Amazon: [link](https://www.amazon.com/gp/product/B08FRL4QDR)

They are a white label version of the same Tuya TRV.  I have a USB version of the CC2652P2 from Tubes. I have successfully installed Zigbee2MQTT on Home Assistant and am able to access the UI.

All of my other devices are ZWave, so this is my first attempt at pairing a Zigbee device. I've followed these [instructions ](https://www.zigbee2mqtt.io/devices/TS0601_thermostat.html)for pairing them. However, the logs don't indicate any connection being made:

Debug 2022-05-08 19:14:36Received MQTT message on 'zigbee2mqtt/bridge/request/permit_join' with data '{"device":null,"time":254,"transaction":"slzvf-1","value":true}'
Info 2022-05-08 19:14:36Zigbee: allowing new devices to join.
Info 2022-05-08 19:14:36MQTT publish: topic 'zigbee2mqtt/bridge/response/permit_join', payload '{"data":{"time":254,"value":true},"status":"ok","transaction":"slzvf-1"}'


I would appreciate any advice to help me get these working! Thanks,
Neil

**Comment by aso824** [comment]
@neil1111 you need to enter into pairing mode on the device after permitting to join. To achieve this, hold central button for a while and release. Device should join after a few seconds.

**Comment by roobbb** [comment]
Hello.

I also have such a device:
model TS0601
manufacturer  _TZE200_hue3yfsn
Labled as EARU Electric TV02 (https://www.aliexpress.com/item/1005003444955449.html)

Is there any chance to set values for schedule or holiday_start_stop?
When I try to set some data as mentioned here https://www.zigbee2mqtt.io/devices/TV02-Zigbee.html#schedule-composite , z2m always tells me "No converter available for ..." undefined, schedule etc. 

There is even a discussion https://github.com/Koenkk/zigbee2mqtt/discussions/11096, but no solution yet as it seems.

Thank you and kind regards
roobbb

**Comment by vladi1234** [comment]
Bug Fix 'holiday_start_stop' for TV02-Zigbee

https://github.com/Koenkk/zigbee-herdsman-converters/pull/4682/files

**Comment by station7** [comment]
> I'm also trying to crack Schedule, but Schedule has a complex protocol unknown to me. I also don't have a Tuya Gateway to listen to.

Hi vladi1234, 

I'd love to be able to edit the schedule of those TV-02 devices as well - even though I use some individual automatic, I would be pleased to have a fallback schedule on the device itself, just in case someone switches the thermostat intentionally or unintentionally to auto mode. I am willing to (co-)sponsor a Tuya gateway for you to listen to - let me know how much you need and how to donate.

**Comment by vladyspavlov** [comment]
Hi @station7,
I am currently working on the schedule function, you can check the progress here https://github.com/Koenkk/zigbee-herdsman-converters/pull/4750

**Comment by kiekerjan18** [comment]
Hi, I am having issues with _TZE200_a4bpgplm TRVs. 

With an HA automation I am setting 3 similar TRV's to a new setpoint. Quite often 1 of the 3 misses this new setpoint and therefor the TRV keeps heating or stays off. 

I've tried many things, but in the logs I see e.g.: Received MQTT message on 'zigbee2mqtt/TRV Achter/set/current_heating_setpoint' with data '15.0'

But for the one missing, I don't see any logging in Z2M. 

For now I've decided to set my TRV via a virtual TRV and switching the real one to turn on or turn off. This doesn't give any missed signals so far.

Any thoughts?
